Moab.Models/Helpers/ExerciseImportHelper.cs

```diff
@@ -20,17 +22,159 @@ public class ExerciseImportHelper
+            exercise.Name = updateCSV[(int)ExerciseCSVColumns.Name];
+            exercise.EasierHint = updateCSV[(int)ExerciseCSVColumns.HintEasier];
+            exercise.HarderHint = updateCSV[(int)ExerciseCSVColumns.HintHarder];
+            exercise.HasRepetitionTarget = ConvertYNtoBool(updateCSV[(int)ExerciseCSVColumns.UnitTarget]);
+            exercise.DateLastUpdated = DateTime.Now;
+        }
+
+        // Converts Y into true and N into false
+        protected bool ConvertYNtoBool(string input)
+        {
+            if (input.ToUpper() == "Y")
+            {
+                return true;
+            }
+            else if (input.ToUpper() == "N")
+            {
+                return false;
+            }
             return false;
         }
 
+        // Converts the input string into a
+        protected List<string[]> SplitCSVInput(string CSVInput)
+        {
+            // Create LineList
+            List<string[]> LineList = new List<string[]>();
+            int iterator = 0;
+            do
+            {
+                string[] line = new string[14];
+                int iteratorNext = CSVInput.IndexOf('\n', iterator);
+                string temp;
+                if (iteratorNext == -1)
+                {
+
+                    temp = CSVInput.Substring(iterator, CSVInput.Length - iterator);
+                }
+                else
+                {
+                    temp = CSVInput.Substring(iterator, iteratorNext - iterator);
+                }
+                line = SplitCSVLine(temp);
+                LineList.Add(line);
+                iterator = ++iteratorNext;
+            }
+            while (iterator > 0);
+            if (LineList.Count == 0)
+            {
+                throw new FormatException("Invalid Format of CSV Input");
+            }
+
+            // Delete Header and Empty Row
+            LineList.RemoveRange(0, 2);
+
+            // Return edited list
+            return LineList;
+        }
+
+        // Splits each line into an array of strings
+        // Helper function to SplitCSVInput
+        // TODO: Implementation for non-fixed number of hints
+        private string[] SplitCSVLine(string CSVLine)
+        {
+            string[] Line = new string[14];
+            int iterator = 0;
+            for (int i = 0; i < 14; i++)
+            {
+                if (CSVLine[iterator] == '\"')
+                {
+                    int iteratorNext = CSVLine.IndexOf('\"', iterator);
+                    if (iteratorNext == -1)
+                    {
+                        throw new FormatException("Invalid Format of CSV Input");
+                    }
+                    string temp = CSVLine.Substring(iterator, iteratorNext - iterator);
+                    Line[i] = temp;
+                    iterator = iteratorNext + 2;
+                }
+                else
+                {
+                    int iteratorNext = CSVLine.IndexOf(',', iterator);
```

Comment:
I suspect that you may want to consider "not finding" the last comma as not an error condition.  This works now because you have a hardcoded number of columns.  What if you don't?

Reply:
Fixed through my re-writing of the SplitCSVLine method